docs/guides/parallel-scraping/parallel-scraping.mdx

```diff
@@ -0,0 +1,209 @@
+
+There are three steps we want to do for the worker processes:
+
+- ensure the default storages do **not** get purged on start, as otherwise we'd lose the queue we prepared
+- get the queue that supports locking from the same location as the parent process
+- initialize a special storage for worker processes so they do not collide with each other
+
+In order, that's what these lines do:
+
+```javascript title="src/parallel-scraper.mjs"
+// Disable the automatic purge on start (step 1)
+// This is needed when running locally, as otherwise multiple processes will try to clear the default storage (and that will cause clashes)
+Configuration.getGlobalConfig().set('purgeOnStart', false);
+
+// Get the request queue from the parent process (step 2)
+const requestQueue = await getOrInitQueue(false);
+
+// Configure crawlee to store the worker-specific data in a separate directory (needs to be done AFTER the queue is initialized when running locally) (step 3)
+const config = new Configuration({
+    storageClientOptions: {
+        localDataDirectory: `./storage/worker-${process.env.WORKER_INDEX}`,
+    },
+});
+```
+
+#### Enabling the request locking experiment, and telling the crawler to use the worker configuration
+
+You might have noticed several lines highlighted in the code above. Those show how you can enable the request locking experiment, as well as how you provide the request queue to the crawler.
+You can read more about the experiment by visiting the [request locking experiment](../experiments/experiments-request-locking) page.
+
+You might have also noticed we passed in a second parameter to the constructor of the crawler, the `config` variable we created earlier.
+This is needed to ensure the crawler uses the worker-specific storages for internal states, and that they do not collide with each other.
+
+#### Why do we use `process.send` instead of `Dataset.pushData`?
+
+Since we use child processes, and each worker process has its own storage space, calling `Dataset.pushData` will not work as we want it to work (each worker would just
+push to it's own personal dataset that is considered the "default" one). Instead, we need to send the data back to the parent process, which has the dataset where we want to
+store the data, in a centralized place.
+
+:::tip Why don't we apply the same logic we did to the request queue to the dataset?
+
+This is a very valid question, but it has a simple answer: since each process tracks its own internal state of how a dataset looks like (when we are scrapping locally),
+the worker processes would get out of sync real fast and would either miss or override data. This is why we need to send the data back to the parent process, which has the dataset
+where we want to store the data, in a centralized place.
+
+Depending on your crawler, this might not be an issue! Each use case has its own quirks, but this is something you should keep in mind when building your scraper.
+
+:::
```

Comment:
Hmm, this sounds like an "us" issue 😬😄 Given your expertise @vladfrangu , how much time would it take to make local Dataset parallel-friendly (I'm assuming the Apify Platform dataset isn't like this...?) I guess it would hurt the cache performance, right?

Reply:
Without fs storage of the dataset metadata, it's most likely impossible (maybe there's a way if we detect we can send data across threads but I do not want to cross that bridge of uncertainty and break who knows what). And given that storing said metadata is *optional* as well..

Realistically, this is an "issue" only with storages that rely on their process, and that aren't centralized. Apify wouldn't have this issue, a DB _probably_ wouldn't have this issue. To be fair, you'd need to store the requests on fs in the first place to have this work, so maybe we..actually enable storing metadata by default and then rely on that?

Also yes, it would possibly tank performance (and I am SCARED of the race conditions that that can cause)

Reply:
Gotcha, no biggie then 👍🏽